AxonDeepSeg/train_network.py

```diff
@@ -75,63 +73,104 @@ def train_model(path_trainingset, path_model, config, path_model_init=None,
     no_valid_images = int(len(os.listdir(path_validation_set)) / 2)
     valid_ids = [str(i) for i in range(no_valid_images)]
 
-    # Loading the Training images and masks in batch
-    train_gen = DataGen(train_ids, path_training_set, batch_size=no_train_images, image_size=image_size,
-                        thresh_indices=thresh_indices)
-    image_train_gen, mask_train_gen = train_gen.__getitem__(0)
 
-    # Loading the Validation images and masks in batch
-    valid_gen = DataGen(valid_ids, path_validation_set, batch_size=no_valid_images, image_size=image_size,
-                        thresh_indices=thresh_indices)
-    image_valid_gen, mask_valid_gen = valid_gen.__getitem__(0)
 
     ###################################################################################################################
     ############################################# DATA AUGMENTATION ##################################################
     ###################################################################################################################
 
-    # Data dictionary to feed into image generator
-    data_gen_args = dict(horizontal_flip=True,
-                         # flipping()[1],
-                         vertical_flip=True,
-                         # preprocessing_function = elastic
-                         # flipping()[0],
-                         # rotation_range = random_rotation()[0],
-                         # width_shift_range = shifting(patch_size, n_classes)[1],
-                         # height_shift_range = shifting(patch_size, n_classes) [0],
-                         # fill_mode = "constant",
-                         # cval = 0
-                         )
+    shifting = config["da-0-shifting-activate"],
+    rescaling = config["da-1-rescaling-activate"]
+    rotation = config["da-2-random_rotation-activate"]
+    elastic = config["da-3-elastic-activate"]
+    flipping = config["da-4-flipping-activate"]
+    gaussian_blur = config["da-5-gaussian_blur-activate"]
```

Comment:
Here, you've implicitly changed the config key from 

`da-5-noise_addition-activate`

to

`da-5-gaussian_blur-activate`

Meaning that all our (and my) old configs will break. A better way to handle this refactor would be to retain the old compatibility as a backup, e.g. something like

```python

if "da-5-gaussian_blur-activate" in config:
    gaussian_blur = config["da-5-gaussian_blur-activate"]
elif "da-5-noise_addition-activate" in config:
    # Preserved for retrocompatibility with old configs
    gaussian_blur = config["da-5-noise_addition-activate"]
end
```

I discovered this because the training notebook failed (so that's another change that's needed, both da-5-gaussian_blur-activate is needed as a key, but also da-6-reflection_border-activate is missing.

Reply:
Fixed in commit(#1041d32)

Reply:
This needs to be fixed in the training guidelines notebook too, as it calls `da-6-reflection_border-activate`

Reply:
Sorry, it's because it's missing there